Add improvements. (#2639)
wso2/docs-is: en/docs/references/concepts/authentication/session-management.md

# OpenID Connect Session Management

The OpenID Connect Session Management [specification](https://openid.net/specs/openid-connect-session-1_0.html)

> defines methodologies to manage user sessions and logout end-users at the authorization server using front-channel communication.

Reviewer: ```suggestion
defines methodologies to manage user sessions and log out end-users at the authorization server using front-channel communication.
```

The OP passes this value to the RP in the authentication response and the RP uses this value 
to monitor end-user session at the OP.

---

## Session Management Endpoints

OpenID Connect uses the following two endpoints for session management. 

1. **check_session_iframe**:
This endpoint supports cross-origin communications for session state information with the client, using the HTML5 postMessage API.
It accepts postMessage requests from the relevant [RP iframe](#rp-iframe) and uses postMessage to post back the login
status of the end-user at the OpenID Provider.

2. **end_session_endpoint**:
This endpoint is responsible for logging out an end-user once the client performs a redirect request.

---
## How does OpenID Connect Session Management work?

The following diagram represents the flow.

![session-management](../../../assets/img/concepts/session-management.png)

- Once the end-user needs to log in to the RP, the RP sends an authentication request to the OP.
- The OP responds back with the **session_state**.
- The RP iframe continuously polls the OP iframe to detect any state changes.

> - The OP iframe responds back with one of the following statuses.

Reviewer: ```suggestion
- The OP iframe responds with one of the following statuses.
```

The status can be either;

- changed: This indicates that user-session is still valid at the OP. RP will continue to poll
             OP iframe to detect any session changes. 
- unchanged: This indicates that the session has changed at the OP.
               This can happen due to user logout, session timeout, or a user logging in from a different client application.
               Upon receipt of the `changed` state, the RP performs re-authentication with prompt=none 
               to obtain the current session state at the OP.
- error: This indicates the message sent was determined by the OP to be malformed.
           Upon receipt of error, the RP must not perform re-authentication with `prompt=none`,

>            so as to not cause potential infinite loops that generate network traffic to the OP, so it directly logged the user out.

Reviewer: ```suggestion
           so as to not cause potential infinite loops that generate network traffic to the OP, it directly logs the user out.
```
